index.js

```diff
@@ -354,7 +353,8 @@
         lookup: function () {
             var editorBody = this.editor.getBody().querySelector('#autocomplete-searchtext');
 
-            if (!editorBody || !editorBody.innerText) {
+            if (!editorBody || !editorBody.innerText || (this.options.delimiter === '#' &&
+                this.editor.container.closest('comment'))) {
```

Comment:
I would suggest to have glossary reference feature flag available from RTF component options and if it's not passed in just not register '#' as delimiter (it even should be possible to do outside of the plugin). Don't think it's a good idea to detect feature based on container (I'm not sure it's gonna work for replies)